github/Project.py

```diff
@@ -0,0 +1,269 @@
+        :rtype: :class:`github.PaginatedList.PaginatedList` of :class:`ProjectCard`
+        """
+        url_parameters = dict()
+        # TODO: support for archived_state parameter when getting cards: all,archived, or not_archived
+        
+        # 'Accept' header required while Github Projects API still in preview mode.
+        headers = { 'Accept': "application/vnd.github.inertia-preview+json" }
+        
+        return github.PaginatedList.PaginatedList(
+            ProjectCard,
+            self._requester,
+            self.url + "/cards",
+            url_parameters,
+            headers
+        )
+        
+    def _initAttributes(self):
+        self._id = github.GithubObject.NotSet
+        self._name = github.GithubObject.NotSet
+        self._url = github.GithubObject.NotSet
+
+    def _useAttributes(self, attributes):
+        if "id" in attributes:  # pragma no branch
+            self._id = self._makeIntAttribute(attributes["id"])
+        if "name" in attributes:  # pragma no branch
+            self._name = self._makeStringAttribute(attributes["name"])
+        if "url" in attributes:  # pragma no branch
+            self._url = self._makeStringAttribute(attributes["url"])
+
+# TODO: remaining ProjectCard properties
+# NOTE: There is currently no current way to get cards "in triage" for a project.
+# https://platform.github.community/t/moving-github-project-cards-that-are-in-triage/3784
+class ProjectCard(github.GithubObject.NonCompletableGithubObject):
+    """
+    This class represents Project Cards. The reference can be found here https://developer.github.com/v3/projects/cards
+    """
+
+    def __repr__(self):
+        return self.get__repr__({"id": self._id.value})
+
+    # content_type is not a property of a returned card, but this property
+    # is consistent with the parameters provided when creating cards, see:
+    # https://developer.github.com/v3/projects/cards/#create-a-project-card
+    @property
+    def content_type(self):
+        """
+        :type: string
+        """
+        if not self.content_url:
+            return None
+        elif "/issues/" in self.content_url:
+            return "Issue"
+        elif "/pulls/" in self.content_url:
+            return "PullRequest"
+        else:
+            return "Unknown"
+            
+    @property
+    def content_url(self):
+        """
+        :type: string
+        """
+        return self._content_url.value
+       
+    @property
+    def id(self):
+        """
+        :type: integer
+        """
+        return self._id.value
+
+    @property
+    def note(self):
+        """
+        :type: string
+        """
+        return self._note.value
+
+    @property
+    def url(self):
+        """
+        :type: string
+        """
+        return self._url.value
+
+    # TODO: get issue from card, not just pull request ("get_content" method?)
+    def get_pullrequest(self):
```

Comment:
Thoughts on this? What do you think of a `get_content( )` method that returns either a PullRequest or an Issue? Note that "note" cards have no content_url, so the only content is the note.

Reply:
GitHub's new v4 GraphQL API has a ProjectCard class with a `content` property, which I will emulate. However I think it is more clear that this is issuing another API call, so I will call the method `get_content`. (Feedback welcome.)